plugins/balloontoolbar/plugin.js

```diff
@@ -176,6 +176,16 @@
 		this._view.destroy();
 	};
 
+	/**
+	 * Refreshes all commands state so buttons state on balloonToolbar are up to date.
+	 */
+	CKEDITOR.ui.balloonToolbar.prototype.refresh = function() {
+		for ( var itemName in this._items ) {
+			var command = this._view.editor.getCommand( this._items[ itemName ].command );
+			command.refresh( this._view.editor, this._view.editor.elementPath() );
```

Comment:
Not all buttons will have commands as our API allows to bind `click` handler instead.